cms/models.py

```diff
@@ -1260,17 +1296,39 @@ def parent(self):
 
     def get_context(self, request, *args, **kwargs):
         preview_context = {}
+        context = {}
 
         if request.is_preview:
             preview_context = {
                 "learner_name": "Anthony M. Stark",
                 "run": self.parent.product.first_unexpired_run,
             }
+        elif self.certificate:
+            # Verify that the certificate in fact is for this same course
+            if self.parent.product != self.certificate.course_run.course:
+                raise Http404()
+
+            context = {
+                "certificate_user": self.certificate.user,
+                "learner_name": self.certificate.user.get_full_name(),
+                "run": self.certificate.course_run,
+            }
+        else:
+            raise Http404()
 
+        # The share image url needs to be absolute
         return {
+            "site_name": settings.SITE_NAME,
+            "share_image_url": urljoin(
+                request.build_absolute_uri("///"),
+                static("images/certificates/share-image.png"),
+            ),
+            "share_image_width": "1665",
+            "share_image_height": "1291",
```

Comment:
suggestion: Should these not be configurable/changeable for CMS or should be fixed ?

Reply:
This isn't going to change for the foreseeable future so I'm not too concerned at this moment